src/lib/components/streams/write-message-modal.svelte

```diff
@@ -1,34 +1,54 @@
-<script lang="ts">
+<script  lang="ts">
     import { FieldType, type WriteMessageForm } from '$lib/app'
     import { selectedChannelBusy, writeMessage } from '$lib/app/streams'
     import type { Input as InputType, SubmitButton } from '$lib/app/types/form'
     import { Form } from '$lib/components'
     import { onDestroy } from 'svelte'
     import { ModalBody, ModalHeader } from 'sveltestrap'
+
     // We have to import Modal this way, otherwise it shouts SSR issues.
     import Modal from 'sveltestrap/src/Modal.svelte'
-
     export let address: string
     export let isOpen: boolean = false
     export let title: string = 'Write a message'
+    export let channelType:string = ''
     export let onModalClose = (..._: any[]): void => {}
     export let onSuccess = (..._: any[]): void => {}
-
     const MAX_LENGTH_TEXTAREA = 100
     const MIN_LENGTH_TEXTAREA = 3
-
     let timeout: NodeJS.Timeout
     let formLoading = false
-    let formInputs: InputType[] = [
+    let formInputPrivate: InputType[] = [
```

Comment:
please keep the forminput the same you can always display payload also for public channels but you must differentiate in 
```
const message = await writeMessage(address, payload, publicPayload, metadata, type, true)
```
once you set `payload` (for private channels) and once you set `publicPayload` (for public channels)